cmd/oras/internal/argument/file_windows_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package argument
+
+import "testing"
+
+func TestParseFileReference(t *testing.T) {
+	type args struct {
+		reference string
+		mediaType string
+	}
+	tests := []struct {
+		name          string
+		args          args
+		wantFilePath  string
+		wantMediatype string
+	}{
+		{"file name and media type in reference", args{"a:b", "c"}, "a", "b"},
+		{"media type in reference", args{":b", "c"}, "", "b"},
+		{"file name and empty media type in reference", args{"a:", "c"}, "a", ""},
+		{"file name in reference", args{"a", "c"}, "a", "c"},
+		{"file name in reference, no default", args{"a:", ""}, "a", ""},
+		{"windows file name in reference with default media type", args{`a:\b`, "c"}, `a:\b`, "c"},
+		{"windows file name and media type in reference", args{`a:\b:c`, "d"}, `a:\b`, "c"},
+		{"file name and media type in reference", args{"a:b:c", "d"}, "a:b", "c"},
```

Comment:
Do we need to test something like:

- `a:b:` (second colon at the end)
- `:a:b` (first colon at the beginning)
- `:` (single colon)
- `::` (double colons)
- `1:\` (first character is not a letter)

They are not valid windows paths though.

Reply:
Yes they are needed. I also comeup with one `ab:\c:d`